src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -500,3 +503,106 @@ export async function refundRequestIsValid(
 
   return { valid: true };
 }
+
+export type SpokePoolEventFilter = {
+  originChainId?: number;
+  destinationChainId?: number;
+  relayer?: string;
+  fromBlock?: number;
+};
+
+export type SpokePoolFillFilter = SpokePoolEventFilter & {
+  repaymentChainId?: number;
+  isSlowRelay?: boolean;
+};
+
+// @description Search for fills recorded by a specific SpokePool.
+// @param chainId Chain ID of the relevant SpokePoolClient instance.
+// @param spokePoolClients Set of SpokePoolClient instances, mapped by chain ID.
+// @param filter  Optional filtering criteria.
+// @returns Array of FillWithBlock events matching the chain ID and optional filtering criteria.
+export async function getFills(
+  chainId: number,
+  spokePoolClients: SpokePoolClients,
+  filter: SpokePoolFillFilter = {}
+): Promise<FillWithBlock[]> {
+  const spokePoolClient = spokePoolClients[chainId];
+  assert(isDefined(spokePoolClient));
+
+  const { originChainId, repaymentChainId, relayer, isSlowRelay, fromBlock } = filter;
+
+  const fills = await filterAsync(spokePoolClient.getFills(), async (fill) => {
+    if (!isDefined(spokePoolClient)) {
+      return false;
+    }
+
+    if (isDefined(fromBlock) && spokePoolClient.latestBlockNumber - fill.blockNumber > fromBlock) {
```

Comment:
i don't understand this math. Are you checking that `fill.blockNumber` is not greater than the spoke pool client's `latestBlockNumber` ? Are you additionally checking that its not less than the `fromBlock`..? A comment here on what you're doing would help me form a review

Reply:
Also a good call. This code has switched between receiving a block offset and an absolute block number, but it settled on an absolute block number because that was most practical. It's of course correct to compare `fill.blockNumber` against `fromBlock`.